modules/react/tabs/lib/TabsItem.tsx

```diff
@@ -0,0 +1,217 @@
+  composeHooks,
+  ExtractProps,
+  ellipsisStyles,
+  EllipsisText,
+} from '@workday/canvas-kit-react/common';
+import {Box} from '@workday/canvas-kit-labs-react/common';
+import {OverflowTooltip} from '@workday/canvas-kit-react/tooltip';
+import {SystemIcon} from '@workday/canvas-kit-react/icon';
+import {useListRegisterItem} from './list';
+import {useRovingFocus} from './cursor';
+import {isSelected, useSelectionItem} from './selection';
+import {useOverflowMeasureItem} from './overflow';
+
+import {TabsModelContext} from './Tabs';
+import {TabsModel} from './useTabsModel';
+export interface TabsItemProps extends ExtractProps<typeof Box, never> {
+  /**
+   * Optionally pass index to tab item. This should be done if `Tabs.Item` components were created
+   * via a `Array::map` function. This index will ensure keyboard navigation works even if items are
+   * inserted out of order.
+   */
+  index?: number;
+  /**
+   * The label text of the Tab.
+   */
+  children: React.ReactNode;
+  /**
+   * The name of the tab. This name will be used in change events and for `initialTab`. Must match
+   * the `name` of the associated tab panel. If this property is not provided, it will default to a
+   * string representation of the the zero-based index of the Tab when it was initialized.
+   */
+  name?: string;
+  /**
+   * Optionally pass a model directly to this component. Default is to implicitly use the same
+   * model as the container component which uses React context. Only use this for advanced use-cases
+   */
+  model?: TabsModel;
+  /**
+   * Optional id. If not set, it will inherit the ID passed to the `Tabs` component and append the
+   * index at the end. Only set this for advanced cases.
+   */
+  id?: string;
+  /**
+   * Part of the ARIA specification for tabs. This attributes links a `role=tab` to a
+   * `role=tabpanel`. This value must be the same as the associated `id` attribute of the tab panel.
+   * This is automatically set by the component and should only be used in advanced cases.
+   */
+  'aria-controls'?: string;
+  /**
+   * Part of the ARIA specification for tabs. Lets screen readers know which tab is active. This
+   * should either be `true` or `undefined` and never `false`. This is automatically set by the
+   * component and should only be used in advanced cases.
+   */
+  'aria-selected'?: boolean;
+  /**
+   * Part of the ARIA specification for tabs. The currently active tab should not have a `tabIndex`
+   * set while all inactive tabs should have a `tabIndex={-1}`
+   */
+  tabIndex?: number;
+  /**
+   * Use `hasIcon={true}` when the Tab item contains an icon. This instructs the tab item to render
+   * children as a flex container to accommodate icons. You will have to use `Tabs.Item.Text` as a
+   * child element if this is set to true.
+   *
+   * @example
+   * <Tabs.Item hasIcon>
+   *   <Tabs.Item.Icon icon={someIcon} />
+   *   <Tabs.Item.Text>Tab Text</Tabs.Item.Text>
+   * </Tabs.Item>
+   */
```

Comment:
I'm wondering if there's any way we can warn the consumer if they don't do this and pass an icon?

Reply:
That could definitely be a dev-only warning, but it might involve extra code and state tracking that is confusing and harder to maintain.

Reply:
How about https://github.com/Workday/canvas-kit/pull/1325/files#diff-12fd6ab47eb13fddf64e87d63f64b95b38acd128ff0e4ed69e4ab0524e782bf4R185-R198

Reply:
awesome!